packages/expo-modules-autolinking/src/reactNativeConfig/reactNativeConfig.ts

```diff
@@ -97,7 +97,7 @@ export async function resolveDependencyConfigAsync(
   const libraryConfig = await loadConfigAsync<RNConfigReactNativeLibraryConfig>(packageRoot);
   const reactNativeConfig = {
     ...libraryConfig?.dependency,
-    ...projectConfig?.dependencies[name],
+    ...(projectConfig?.dependencies ? projectConfig?.dependencies[name] : {}),
```

Comment:
```suggestion
    ...projectConfig?.dependencies?.[name],
```

that would be good enough